src/readability/opportunities/guidance-handler.js

```diff
@@ -0,0 +1,170 @@
+/*
+ * Copyright 2025 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+import { ok, notFound } from '@adobe/spacecat-shared-http-utils';
+
+/**
+ * Maps Mystique readability suggestions to opportunity format
+ * @param {Array} mystiquesuggestions - Array of suggestions from Mystique
+ * @returns {Array} Array of suggestions for opportunity
+ */
+function mapMystiqueSuggestionsToOpportunityFormat(mystiquesuggestions) {
+  return mystiquesuggestions.map((suggestion, index) => {
+    const suggestionId = `readability-opportunity-${suggestion.pageUrl || 'unknown'}-${index}`;
```

Comment:
suggestion ID should be auto generated UUID, not sure why it's a manually constructed string here